### ISSUE TITLE
Influx 0.13.0 with Riemann 0.2.10 - database is required

### ISSUE THREAD

**Issue by rudijs** [riemann/riemann#715]
Hi,

I'm having an issue with InfluxDB version: 0.13.0 and Riemann version: 0.2.10 that I'm stuck with and would like to request help.

Not sure if I'm doing it wrong or if it's an issue/bug.

The problem is InfluxDB is returning `body "{\"error\":\"database is required\"}\n"}` for each event sent to it from my Riemann instance.

I have all this running locally inside Docker containers, pretty sure I have it configured correctly.

Here's the details:

InfluxDB running with a 'riemann' database created.

I can use `curl` and send events no problem.

```
Connected to http://localhost:8086 version 0.13.0
InfluxDB shell version: 0.13.0
> use riemann
Using database riemann
> show databases
name: databases
---------------
name
_internal
riemann
```

Send an event:

```
curl -i -XPOST 'http://192.168.33.100:8086/write?db=riemann' --data-binary 'cpu_load_short,host=server01,region=us-west value=0.64'
HTTP/1.1 204 No Content
Request-Id: 4c7523c5-65b6-11e6-85b9-000000000000
X-Influxdb-Version: 0.13.0
Date: Fri, 19 Aug 2016 02:40:32 GMT
```

InfluxDB logs OK:

```
influxdb    | [httpd] 2016/08/19 02:40:32 192.168.33.1 - - [19/Aug/2016:02:40:32 +0000] POST /write?db=riemann HTTP/1.1 204 0 - curl/7.43.0 4c7523c5-65b6-11e6-85b9-000000000000 729.296µs
```

Can connect and select from the database no worries:

```
# influx -database riemann
Visit https://enterprise.influxdata.com to register for updates, InfluxDB server management, and monitoring.
Connected to http://localhost:8086 version 0.13.0
InfluxDB shell version: 0.13.0
> select * from cpu_load_short
name: cpu_load_short
--------------------
time                    host            region  value
1471574432257772661     server01        us-west 0.64
```

So I've tried a few different configs but taking a step back now and trying to get the most simple as possible working.

I know in production the influxdb function will be wrapped with batch and async-queue!, but for now it's at it's simplest form:

```
; log to stdout
(logging/init)

; Listen on the local interface over TCP (5555), UDP (5555), and websockets
; (5556)
(let [host "0.0.0.0"]
 (tcp-server {:host host})
 (udp-server {:host host})
 (ws-server  {:host host}))

; Expire old events from the index every 5 seconds.
; Events (Services) are added to the index, as a service, by mapping the :host and :service properties from the event. 
(periodically-expire 5)


(let [index (index)]
; Inbound events will be passed to these streams:
 (streams
   (default :ttl 60
    ; Index all events immediately.
    ; index is a function that sends events to Riemann's Index
     index

    ; Log expired events.
     #(info %)
     (influxdb { :version :0.9 :host "192.168.33.100" :port 8086 :db "riemann"}))))
```

So events generated by riemann itself, plus any I send with a client, are sent to influxdb.

This is the full response:

```
INFO [2016-08-19 02:38:51,437] Thread-6 - riemann.config - #riemann.codec.Event{:host eek, :service riemann server ws 0.0.0.0:5556 in rate, :state ok, :description nil, :metric 0.0, :tags nil, :time 1471574331253/1000, :ttl 20}
WARN [2016-08-19 02:38:51,444] Thread-6 - riemann.streams - riemann.influxdb$influxdb_9$stream__10987@32c9498 threw
clojure.lang.ExceptionInfo: clj-http: status 400 {:orig-content-encoding "gzip", :trace-redirects ["http://192.168.33.100:8086/write"], :request-time 6, :status 400, :headers {"Connection" "close", "Content-Length" "57", "Date" "Fri, 19 Aug 2016 02:38:51 GMT", "X-Influxdb-Version" "0.13.0", "Request-Id" "105d86db-65b6-11e6-85b4-000000000000", "Content-Type" "application/json"}, :body "{\"error\":\"database is required\"}\n"}
        at clj_http.client$wrap_exceptions$fn__9677.invoke(client.clj:183)
        at clj_http.client$wrap_accept$fn__9829.invoke(client.clj:536)
        at clj_http.client$wrap_accept_encoding$fn__9835.invoke(client.clj:550)
        at clj_http.client$wrap_content_type$fn__9824.invoke(client.clj:525)
        at clj_http.client$wrap_form_params$fn__9906.invoke(client.clj:697)
        at clj_http.client$wrap_nested_params$fn__9923.invoke(client.clj:726)
        at clj_http.client$wrap_method$fn__9876.invoke(client.clj:641)
        at clj_http.cookies$wrap_cookies$fn__8113.invoke(cookies.clj:124)
        at clj_http.links$wrap_links$fn__8969.invoke(links.clj:51)
        at clj_http.client$wrap_unknown_host$fn__9932.invoke(client.clj:742)
        at clj_http.client$post.doInvoke(client.clj:857)
        at clojure.lang.RestFn.invoke(RestFn.java:423)
        at riemann.influxdb$influxdb_9$stream__10987.invoke(influxdb.clj:176)
        at riemann.streams$default$stream__5044$fn__5055.invoke(streams.clj:1353)
        at riemann.streams$default$stream__5044.invoke(streams.clj:1353)
        at riemann.core$stream_BANG_$fn__5727.invoke(core.clj:19)
        at riemann.core$stream_BANG_.invoke(core.clj:18)
        at riemann.core$instrumentation_service$measure__5736.invoke(core.clj:57)
        at riemann.service.ThreadService$thread_service_runner__3283$fn__3284.invoke(service.clj:71)
        at riemann.service.ThreadService$thread_service_runner__3283.invoke(service.clj:70)
        at clojure.lang.AFn.run(AFn.java:22)
        at java.lang.Thread.run(Thread.java:745)
```

The 400 response log from InfluxDB:

```
influxdb    | [httpd] 2016/08/19 02:56:00 192.168.33.100 - - [19/Aug/2016:02:56:00 +0000] POST /write HTTP/1.1 400 57 - Apache-HttpClient/4.4.1 (Java/1.7.0_111) 75df667a-65b8-11e6-85c5-000000000000 288.878µs
```

So in summary....

I think I have it configured correctly, can you see where I've gone wrong?

The only difference I can detect between my `curl` command and the `riemann` sent http request is the lack of `?db=riemann` in the logs.

Any help/direction/advice here would be greatly appreciated.

Thanks!

**Comment by pradeepchhetri** [comment]
Hi @rudijs 

I don't think, Riemann 0.2.10 supports InfluxDB 0.13.0. I am sure upgrading to Riemann 0.2.11 will fix your issue.

For your reference, here is the how it was implemented in 0.2.10:
https://github.com/riemann/riemann/blob/0.2.10/src/riemann/influxdb.clj#L150

And in 0.2.11:
https://github.com/riemann/riemann/blob/0.2.11/src/riemann/influxdb.clj#L158

Thanks !!

**Comment by rudijs** [comment]
@pradeepchhetri Hey Pradeep - ah OK.

My bad I wasn't using the latest version.

I think I cut'n'paste some docker build commands without realizing the version wasn't latest.

Thank you kindly for taking the time to point this out.

A quick Docker rebuild here and it's working fine - sweet!

**Comment by pradeepchhetri** [comment]
@rudijs Glad that it worked.
